doc/nyan/nyan.md

```diff
@@ -25,68 +46,149 @@ part of a nyanspec describes a simple unit-type in **openage**.
+of this type in nyan data file :
+
+    +UNITTYPE VILLAGER {
+	    health_points = 10,
+	    speed = 1,
+	    can_build = [BARRACKS, TOWNCENTER]
+    }
+    
+In the current section we will continue to provide some nyan data snippets as part of
+the nyanspec examples. The actual syntax of nyan data files will be detailed in the
+following sections. Let us focus on nyanspec for now.
 
-Each attribute must have a name and a type. An attribute is defined by its name,
-followed by a colon and its type. Allowed types are all nyan types and nyan
-built-in types. Possible built-in types are:
+As you could see, each attribute must have a name and a type. An attribute is
+defined by its name, followed by a colon and then its type. Allowed types are all
+nyan types and nyan built-in types. Possible built-in types are:
 
 - `int`:  integers
 - `float`:  floating point numbers
 - `bool`:  boolean values, either true or false
 - `string`:  character sequences
 
-Further an attribute can be a finite set of nyan objects, which is specified
-with the `set` keyword. It is important to remember that we are really talking
-of sets and not of bags. Therefore no duplicates are allowed within a set and no
-order is guaranteed. For each attribute a default value can be specified after
-the attribute type separated by an equals sign.
+Furthermore, an attribute type can be a finite set of nyan objects of one type.
+A set attribute is simply specified with the `set` keyword. The type of the set is
+specified within the parenthesis following the `set` keyword. It is important to
+remember that we are really talking of sets and not of bags. Therefore no duplicates
+are allowed within a set and no order is guaranteed. 
+
+Finally, for each attribute a default value can be specified after the attribute type
+separated by an equals sign.
+
+**Things to remember (tl;dr)**:
 
-### Dynamic attributes
+- nyan types are *exclusively* defined in the nyanspec file
+- In the nyan data file, nyan objects are *instanciated from known types*. The nyan syntax
+will be detailed in the following sections
+- A set attribute can contain a *set* of objects of *one* given type.
 
-Sometimes there are cases where nyan types need dynamic attributes.
-Imagine a unit in **openage** has multiple abilities. This would lead to the
-following definition:
+### nyanspec - Dynamic attributes
 
+An object can extend a nyan type if the nyanspec of this object type contains an ellipis.
+For types with ellipis, the nyanspec simply specifies the minimum set of attributes for any
+object of this type. Additionnal *dynamic attributes* can be specified in the nyan data file
+when an object is instanciated.
+
+Imagine a villager unit in **openage** has multiple abilities. This would lead to the
+following (not realistic) example:
+
+    # nyanspec
     ABILITY {
-	    name : string
+	    name : string	# Regular attribute specification
+	    ...			# Dynamic attributes are allowed for this type
```

Comment:
This `...` is driving me nuts, we have to come up with a better alternative. Nobody will understand that, everybody will asume that the example will just go on.